src/main/scala/optics/poly/GenLens.scala

```diff
@@ -1,283 +1,269 @@
-      }._2.copy(value)
+      Select.overloaded(from, "copy", Nil, NamedArg(field, to) :: Nil)
     }
 
-    getter.unseal match {
-      case Function(param :: Nil, body) =>
-        typeOf[S].classSymbol.map { cls =>
-          '{
-            val setter = (t: T) => (s: S) => ${
-              tree(cls, body)(('s).unseal, ('t).unseal).seal.cast[S]
-            }
-            Lens.apply($getter, setter)
-          }
-        }.getOrElse {
-          qctx.error("Unsupported syntax. Please explicitly use a concrete class.")
-          '{???}
+    def constructLensExpr(field: String): Expr[Lens[From, To]] = 
+      '{
+        val setter = (to: To) => (from: From) => ${
+          constructSetterBody(field, '{from}.asTerm, '{to}.asTerm).asExprOf[From]
         }
-      case err =>
-        qctx.error(s"Unsupported syntax. Example: `GenLens[Address](_.streetNumber)`, ${err.show}")
+        
+        val getter = (from: From) => ${ 
+          constructGetterBody(field, '{from}.asTerm).asExprOf[To] 
+        }
+        Lens.apply(getter, setter)
+      }
+
+    getterExpr.asTerm match {
+      case ExpectedLambdaFunction(fieldName) =>
+        TypeRepr.of[From].classSymbol match {
+          case Some(fromClass) if fromClass.flags.is(Flags.Case) => constructLensExpr(fieldName)
+          case Some(fromClass) => report.error(s"Expecting a case class in the 'From' position; found $fromClass"); '{???}
+          case None => report.error(s"Expecting a concrete case class in the 'From' position; cannot reify type ${summon[Type[From]]}"); '{???}
+        }
+      case term =>
+        report.error(s"Expecting a lambda function that directly accesses a field. Example: `GenLens[Address](_.streetNumber)`")
         '{???}
     }
   }
 
-  def apply[S] = new MkGenLens[S]
+  def apply[From] = new MkGenLens[From]
 
-  inline def uncurried[S, T](inline get: (S => T)): Lens[S, T] = ${ GenLens.impl('get )}
+  inline def uncurried[From, To](inline get: (From => To)): Lens[From, To] = 
+    ${ GenLens.impl('get) }
 
-  class MkGenLens[S] {
-    inline def apply[T](inline get: (S => T)): Lens[S, T] = ${ GenLens.impl('get) }
+  class MkGenLens[From] {
+    inline def apply[To](inline get: (From => To)): Lens[From, To] = 
+      ${ GenLens.impl('get) }
   }
 
 }
 
 object Focus {
 
-  def impl[S: Type, T](getter: Expr[S => T])(using qctx: QuoteContext, ttt: Type[T]) = {
+  def impl[From: Type, To: Type](getterExpr: Expr[From => To])(using quotes: Quotes): Expr[Lens[From, To]] = {
+    import quotes.reflect._
+  
 
-    import qctx.tasty._
-    import util._
+    ///////////// DOMAIN //////////////////
+    case class ParseParams(argName: String, argType: TypeRepr, lambdaBody: Term)
 
-    object Function {
-      def unapply(t: Term): Option[(Term)] = t match {
-        case Inlined(None, Nil, lambda @ Lambda(params, body)) => Some((lambda))
-        case _ => None
-      }
-    }
+    enum FocusAction {
+      case Field(name: String, from: TypeRepr, to: TypeRepr)
```

Comment:
Do you plan that we add one constructor per special method in the path like `?`, `*`, etc?